core/src/server/resource_limiting.rs

```diff
@@ -169,8 +169,12 @@ pub struct ResourceGuard {
 
 impl Drop for ResourceGuard {
 	fn drop(&mut self) {
-		for (sum, claimed) in self.totals.lock().iter_mut().zip(self.units) {
-			*sum -= claimed;
-		}
```

Comment:
To my eyes this does not look slow; the lock should be held for a **really** short time here.